homeassistant/components/life360/__init__.py

```diff
@@ -2,121 +2,20 @@
-    CONF_SHOW_AS_STATE,
-    CONF_WARNING_THRESHOLD,
-    DEFAULT_OPTIONS,
-    DOMAIN,
-    LOGGER,
-    SHOW_DRIVING,
-    SHOW_MOVING,
-)
+from .const import DEFAULT_OPTIONS, DOMAIN
 from .coordinator import Life360DataUpdateCoordinator, MissingLocReason
 
 PLATFORMS = [Platform.DEVICE_TRACKER, Platform.BUTTON]
-
-CONF_ACCOUNTS = "accounts"
-
-SHOW_AS_STATE_OPTS = [SHOW_DRIVING, SHOW_MOVING]
-
-
-def _show_as_state(config: dict) -> dict:
-    if opts := config.pop(CONF_SHOW_AS_STATE):
-        if SHOW_DRIVING in opts:
-            config[SHOW_DRIVING] = True
-        if SHOW_MOVING in opts:
-            LOGGER.warning(
-                "%s is no longer supported as an option for %s",
-                SHOW_MOVING,
-                CONF_SHOW_AS_STATE,
-            )
-    return config
-
-
-def _unsupported(unsupported: set[str]) -> Callable[[dict], dict]:
-    """Warn about unsupported options and remove from config."""
-
-    def validator(config: dict) -> dict:
-        if unsupported_keys := unsupported & set(config):
-            LOGGER.warning(
-                "The following options are no longer supported: %s",
-                ", ".join(sorted(unsupported_keys)),
-            )
-        return {k: v for k, v in config.items() if k not in unsupported}
-
-    return validator
-
-
-ACCOUNT_SCHEMA = {
-    vol.Required(CONF_USERNAME): cv.string,
-    vol.Required(CONF_PASSWORD): cv.string,
-}
-CIRCLES_MEMBERS = {
-    vol.Optional(CONF_EXCLUDE): vol.All(cv.ensure_list, [cv.string]),
-    vol.Optional(CONF_INCLUDE): vol.All(cv.ensure_list, [cv.string]),
-}
-LIFE360_SCHEMA = vol.All(
-    vol.Schema(
-        {
-            vol.Optional(CONF_ACCOUNTS): vol.All(cv.ensure_list, [ACCOUNT_SCHEMA]),
-            vol.Optional(CONF_CIRCLES): CIRCLES_MEMBERS,
-            vol.Optional(CONF_DRIVING_SPEED): vol.Coerce(float),
-            vol.Optional(CONF_ERROR_THRESHOLD): vol.Coerce(int),
-            vol.Optional(CONF_MAX_GPS_ACCURACY): vol.Coerce(float),
-            vol.Optional(CONF_MAX_UPDATE_WAIT): cv.time_period,
-            vol.Optional(CONF_MEMBERS): CIRCLES_MEMBERS,
-            vol.Optional(CONF_PREFIX): vol.Any(None, cv.string),
-            vol.Optional(CONF_SCAN_INTERVAL): cv.time_period,
-            vol.Optional(CONF_SHOW_AS_STATE, default=[]): vol.All(
-                cv.ensure_list, [vol.In(SHOW_AS_STATE_OPTS)]
-            ),
-            vol.Optional(CONF_WARNING_THRESHOLD): vol.Coerce(int),
-        }
-    ),
-    _unsupported(
-        {
-            CONF_ACCOUNTS,
-            CONF_CIRCLES,
-            CONF_ERROR_THRESHOLD,
-            CONF_MAX_UPDATE_WAIT,
-            CONF_MEMBERS,
-            CONF_PREFIX,
-            CONF_SCAN_INTERVAL,
-            CONF_WARNING_THRESHOLD,
-        }
-    ),
-    _show_as_state,
-)
-CONFIG_SCHEMA = vol.Schema(
-    vol.All({DOMAIN: LIFE360_SCHEMA}, cv.removed(DOMAIN, raise_if_present=False)),
-    extra=vol.ALLOW_EXTRA,
-)
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
```

Comment:
Why was there still a config schema in the first place? We should raise issues to deprecate this properly

Reply:
Because it did not have config flow when first implemented. It only had YAML configuration. To avoid a breaking change, YAML configuration was slowly deprecated. This is the last step in that process.

Reply:
Yes but removing this will cause errors on peoples startup iirc. The deprecation period is 6 months, so I think it would be better that we raise an issue where we highlight that people should remove this configuration from their yaml. That's what we now do with all integrations that remove YAML.

(For context, The yaml for withings was only used to import oauth2 details, not for setting up the integration itself. Even in that case I now raise an issue for the next couple of months so people have time to act on it)

Reply:
`cv.removed` had been in place for a long time. (I don't know if `cv.config_entry_only_config_schema` existed when I first deprecated YAML configuraiton.) I'm just removing all the old schemas that aren't needed anymore.

Reply:
It's been deprecated for over 1.5 years. See https://github.com/home-assistant/core/pull/72461

Reply:
I'll raise this at some other people to see what they think.

Reply:
I think it's fine to remove it as an error log entry was created for 1.5 years.

@pnbruckner Can you please check if `async_setup` is still needed. From my understanding, it can be removed, and if we can remove it, we can completely remove `CONFIG_SCHEMA`.

Reply:
Done